### PR TITLE
Vomit no longer stuns

### PR DESCRIPTION
**Pull request by HMBGERDO**
## What Does This PR Do

This PR removes stun from vomit proc, instead it confuses and slowdowns you, if its blood vomit(IB), it knockouts you.

## Why It's Good For The Game

Random stuns while two IBs, random stuns while toxin damage is bad.

Having no random stuns is good for me so i think this PR should be fine.

## Testing

Compiled, got damage

## Changelog
:cl:
tweak: Vomit no longer stuns, instead it makes you Confused and Knockouts you if its vomit from IB.
/:cl:

### CODE REVIEW

**Comment by matttheficus** [comment]
Unsure how I feel about this.

We already have a rough time with people in DEEP CRIT being able to keep fighting for what feels like an eternity. If youre sub-100 HP you dont deserve to keep fighting without possible downsides IMO.

The easiest way to avoid getting RNG maluses is to just never get there in the first place.

**Comment by HMBGERDO** [comment]
Okay so, updated it
Now if you have -50 - 0 health, you recieve lesser crit, you get knockdown instead of weaken, you get confused instead of stuns
If you have less than -50 health, your life becomes pain, you have more chance to get stunned or paralysed.

It makes crit lighter at start and harder at end, so unless you get 150+ damage you're safe from RNG.

File: code/datums/diseases/critical.dm [hunk 1]
```diff
@@ -71,7 +72,8 @@
 				if(prob(5))
 					to_chat(affected_mob, "<span class='danger'>You feel horrible!</span>")
 				if(prob(5))
-					affected_mob.emote(pick("faint", "collapse", "groan"))
+					affected_mob.emote(pick("groan"))
```
Comment (Henri215) [inline code/datums/diseases/critical.dm:75]:
```suggestion
					affected_mob.emote("groan")
```

File: code/datums/diseases/critical.dm [hunk 2]
```diff
@@ -125,7 +127,8 @@
 					to_chat(affected_mob, "<span class='danger'>Your heart stops beating!</span>")
 					affected_mob.AdjustLoseBreath(6 SECONDS)
 				if(prob(5))
-					affected_mob.emote(pick("faint", "collapse", "groan"))
+					affected_mob.emote(pick("groan"))
```
Comment (Henri215) [inline code/datums/diseases/critical.dm:130]:
```suggestion
					affected_mob.emote("groan")
```

**Comment by HMBGERDO** [comment]
PR was splitted into vomit stun change and crit stuns changes

File: code/modules/mob/living/carbon/carbon.dm [hunk 3]
```diff
@@ -101,14 +101,18 @@
 		return FALSE
 
 	if(stun)
-		Stun(8 SECONDS)
+		if(blood)
+			KnockDown(10 SECONDS)
+		AdjustConfused(8 SECONDS)
+		Slowed(8 SECONDS)
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/carbon.dm:107]: slowed needs a second parameter for the amount it should slow you down by.

File: code/modules/mob/living/carbon/carbon.dm [hunk 4]
```diff
@@ -101,14 +101,18 @@
 		return FALSE
 
 	if(stun)
```
Comment (hal9000PR) [inline code/modules/mob/living/carbon/carbon.dm:103]: would you change the name of the parameter (also set the bools to TRUE/FALSE while you are at it)

**Review by hal9000PR — CHANGES_REQUESTED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**